src/qt/bitcoingui.cpp

```diff
@@ -586,7 +584,10 @@ void BitcoinGUI::setClientModel(ClientModel *_clientModel, interfaces::BlockAndH
         createTrayIconMenu();
 
         // Keep up to date with client
-        updateNetworkState();
+        setNetworkActive(m_node.getNetworkActive());
+        connect(connectionsControl, &GUIUtil::ClickableLabel::clicked, [this] {
+            m_network_context_menu->exec(QCursor::pos());
```

Comment:
`QMenu::exec` is blocking. If you want to keep this approach then you could declare a local `QMenu` instance instead - this way you can ditch the class member and the call to `.clear()`.

Reply:
@promag 

Thanks!

> `QMenu::exec` is blocking.

My bad! [Fixed](https://github.com/bitcoin-core/gui/pull/309#issuecomment-835154862).

> If you want to keep this approach then you could declare a local `QMenu` instance instead - this way you can ditch the class member and the call to `.clear()`.

How the lifetime of local `QMenu` instances could be managed in a proper way?